features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -113,13 +202,71 @@ public void deleteDeviceConfig(long id) {
         deviceConfigDao.delete(id);
     }
 
+    @Override
+    public Response downloadDeviceConfig(@PathParam("id") long id) {
+        throw new UnsupportedOperationException("downloadDeviceConfig not yet implemented.");
+    }
+
+    @Override
+    public Response downloadDeviceConfigs() {
+        throw new UnsupportedOperationException("downloadDeviceConfigs not yet implemented.");
+    }
+
+    private Criteria getCriteria(
+        Integer limit,
+        Integer offset,
+        String orderBy,
+        String order,
+        Integer ipInterfaceId,
+        String configType,
+        Long createdAfter,
+        Long createdBefore
+    ) {
+        var criteriaBuilder = new CriteriaBuilder(DeviceConfig.class);
+
+        if (limit != null) {
+            criteriaBuilder.limit(limit);
+        }
+
+        if (offset != null) {
+            criteriaBuilder.offset(offset);
+        }
+
+        if (orderBy != null) {
+            criteriaBuilder.orderBy(orderBy, "asc".equals(order));
+        }
+
+        if (ipInterfaceId != null) {
+            criteriaBuilder.eq("ipInterface.id", ipInterfaceId);
+        }
+
+        if (StringUtils.isNoneBlank(configType)) {
+            criteriaBuilder.ilike("configType", configType);
+        }
+
+        if (createdAfter != null) {
+            criteriaBuilder.ge("createdTime", new Date(createdAfter));
+        }
+
+        if (createdBefore != null) {
+            criteriaBuilder.le("createdTime", new Date(createdBefore));
+        }
+
+        return criteriaBuilder.toCriteria();
+    }
+
     private static DeviceConfigDTO deviceConfigDto(DeviceConfig deviceConfig) {
         return new DeviceConfigDTO(
-                deviceConfig.getId(),
-                deviceConfig.getIpInterface().getId(),
-                deviceConfig.getConfig(),
-                deviceConfig.getEncoding(),
-                deviceConfig.getCreatedTime()
+            deviceConfig.getId(),
+            deviceConfig.getIpInterface().getId(),
+            InetAddressUtils.str(deviceConfig.getIpInterface().getIpAddress()),
```

Comment:
We should check, if Hibernate does another select for each `IpInterface` separately. Maybe some `eager` annotation is necessary.

Reply:
Good point, will take a look. Also could be resolved by doing a query as you suggested below.

Reply:
May just need a different query, to get more data from node and ipinterface tables, am working on this.

Reply:
It does look like Hibernate makes separate select calls for each `IpInterface`. I will test with `EAGER` annotation. We will almost certainly end up making additional changes once UX is more finalized and we do more testing, so may make sense to wait until then to make more changes.